### PR TITLE
fix `_owner` nullability in `routers.dart`

### LINKED ISSUE #84769: Null check operator used on a null value from`routes.dart`
https://github.com/flutter/flutter/blob/9413cda3d51086bf92f2dfbd120b9c3d03399bc3/packages/flutter/lib/src/widgets/routes.dart#L447

This code:
  void remove() {
    ```_owner!.removeLocalHistoryEntry(this); ``` =>this should be ```_owner?.removeLocalHistoryEntry(this); ```
    assert(_owner == null);
  }
  
  This issue has appeared when :
 ``` 
 PersistentBottomSheetController? persistentBottomSheetController;
  ...
  persistentBottomSheetController = Scaffold.of(context).showBottomSheet<void>((context) =>return Text('')));
...
  persistentBottomSheetController?.close(); /* if bottom sheet is closed by dragging down this error is appears*/
  ```

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/84769

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hi @TahaTesser thanks for the fix! This should have a test. Also, can you rebase your branch? That should resolve the failing checks from an infra issue.

**Comment by Piinks** [comment]
(triage) Would you like to continue with this change @TahaTesser ?

**Comment by TahaTesser** [comment]
@Piinks 
Thanks for the review, I would love to make the changes soon)

**Comment by TahaTesser** [comment]
@Piinks 
Apologies for the delay, I had other commitments,
I have added a test and widget class for the test and rebased the branch
If the `_owner` is not nullable in `remove()` - test fails (as expected) and `_owner` is nullable  - test passes (as expected) as the change here is to make `_owner` nullable.

Thank you so much for the review

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 8.
View them at https://flutter-gold.skia.org/cl/github/84840

File: packages/flutter/test/widgets/routes_test.dart [hunk 1]
```diff
@@ -1700,8 +1700,8 @@ void main() {
       final WidgetWithLocalHistoryState state = tester.state(find.byType(WidgetWithLocalHistory));
       state.addLocalHistory();
       // Waits for modal route to update its internal state;
-      await tester.pump();
 
+      await tester.pump();
```
Comment (goderbauer) [inline packages/flutter/test/widgets/routes_test.dart:1704]: nit: revert this change here? This line belongs to the comment above and there shouldn't be a blank line in between.

File: packages/flutter/test/widgets/routes_test.dart [hunk 2]
```diff
@@ -1712,6 +1712,25 @@ void main() {
       await tester.pump();
       expect(tester.takeException(), null);
     });
+    testWidgets('child with no local history can be disposed', (WidgetTester tester) async {
```
Comment (goderbauer) [inline packages/flutter/test/widgets/routes_test.dart:1715]: nit: add a blank line before this one.

File: packages/flutter/test/widgets/routes_test.dart [hunk 3]
```diff
@@ -1712,6 +1712,25 @@ void main() {
...
+      await tester.pumpWidget(const MaterialApp(
+        home: WidgetWithNoLocalHistory(),
+      ));
+
+      final WidgetWithNoLocalHistoryState state = tester.state(find.byType(WidgetWithNoLocalHistory));
+      state.addLocalHistory();
+      // Waits for modal route to update its internal state;
+
```
Comment (goderbauer) [inline packages/flutter/test/widgets/routes_test.dart:1723]: nit: remove this blank line or place it before the comment

**Review by goderbauer — APPROVED** [review]
LGTM after nits are resolved.

**Comment by TahaTesser** [comment]
@goderbauer 
Thanks so much for the review,, made the changes https://github.com/flutter/flutter/pull/84840/commits/444045ba07091666a4da28887065df8ac455bcb7

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**